public/content/lessons/2017/chain-rule-and-product-rule/index.mdx

```diff
@@ -7,4 +7,355 @@ video: YG15m2VwSjA
+
+<Figure
+  image="figures/product-rule-example-step-2.svg" 
+/>
+
+This line of reasoning works for any two functions.
+
+<Figure
+  image="figures/product-rule-general.svg" 
+  caption="Generic Product Rule where $f(x) = g(x)h(x)$."
+/>
+
+A common mnemonic for the product rule is to say in your head "left d right, right d left". In this example, $\sin(x) \cdot x^2$, "left d right" means you take the left function, in this case $g(x) = \sin(x)$, times the derivative of the right, $h(x) = x^2$, which gives $2x$. Then you add "right d left": the right function, $x^2$, times the derivative of the left, $\cos(x)$.
+
+Out of context, this feels like kind of a strange rule, but when you think of this adjustable box you can actually see how those terms represent slivers of area. "Left d right" is the area of this bottom rectangle, and “right d left” is the area of this rectangle on the right.
+
+## Constant multiplication
+
+By the way, I should mention that if you multiply by a *constant*, say $2 \cdot \sin(x)$, things end up much simpler. The derivative is just that same constant times the derivative of the function, in this case $2 \cdot \cos(x)$. I’ll leave it to you to pause and ponder to verify that this makes sense. <!--TODO: link to question from last lesson -->
+
+<Accordion title="Explanation">
+
+One way to verify that this is the case is to visualize a constant multiplied by a function as the area of the rectangle with width equal to the function and height equal to the constant. 
+
+<Figure
+  image="figures/constant-multiplication-area.svg" 
+/>
+
+Then, introducing a small nudge $dx$ to the input of the function, you can measure the change in the area highlighted in yellow below.
+
+<Figure
+  image="figures/constant-multiplication-area-df.svg" 
+/>
+
+In the case of $g(x) = \sin(x)$ and $c = 2$ we know that $d(g(x)) = \cos(x) \cdot dx$. This gives us a change in area of $df = 2 \cdot \cos(x) \cdot dx$. Solving for $\frac{df}{dx}$ we get that the derivative is $2\cos(x)$.
+
+</Accordion>
+
+## Chain rule
+
+Aside from addition and multiplication, the other common way to combine functions that comes up all the time is function composition. For example, let’s say we take the function $x^2$, and shove it inside $\sin(x)$ to get a new function, $\sin(x^2)$. Or, in other words, the output of the function $x^2$ gets fed as input to sine function.
+
+What’s the derivative of this new function?
+
+Here I’ll choose yet another way to visualize things, just to emphasize that in creative math, we have lots of options. I’ll put up three number lines. The top one will hold the value of $x$, the second one will represent the value of $x^2$, and the third line will hold the value of $\sin(x^2)$. 
+
+<Figure
+  image="figures/chain-rule-x-1-2.svg" 
+/>
+
+That is, the function $x^2$ gets you from line $1$ to line $2$, and the function sine gets you from line $2$ to line $3$.
```

Comment:
This particular image feels like it requires more explanation to really get the hang of. (The question immediately after is great for catching any lack of understanding, but it currently feels like the question is too difficult without a bit more guidance on what the image is showing.)

It might be enough to just show a video clip of the 2nd and 3rd number lines responding to changes in the value of x.

Or maybe it would be useful to walk through, in concrete numbers, why the number line values are what they are. Something like...

"In the image, I'm showing an x value of 0.5 on the first number line. So the second number line, which just displays x^2, is showing the number 0.25. The third number line shows sin(x^2), which is really just the sine of the previous value, so sin(0.25) ≈ 0.247."